besu/src/main/java/org/hyperledger/besu/cli/config/NativeRequirement.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright contributors to Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.cli.config;
+
+import org.hyperledger.besu.evm.precompile.AltBN128PairingPrecompiledContract;
+import org.hyperledger.besu.evm.precompile.BLS12PairingPrecompiledContract;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import ethereum.ckzg4844.CKZG4844JNI;
+
+/**
+ * Record type to encapsulate the result of native library loading
+ *
+ * @param success boolean indicating library loading success or failure.
+ * @param libname string indicating the required library name.
+ * @param errorMessage Optional error message suitable to log.
+ */
+public record NativeRequirement(Boolean success, String libname, Optional<String> errorMessage) {
```

Comment:
```suggestion
public record NativeRequirementLoadingResult(Boolean success, String libname, Optional<String> errorMessage) {
```

Reply:
I refactored NativeRequirement to be a standalone class which provides NativeRequirementResult records :+1: